### ISSUE TITLE
When supplying incorrect type in column, default comparison is used whenever 0 is returned from sortFn.

### ISSUE THREAD

**Issue by SethRogers7420** [xaksis/vue-good-table#582]
### Issue Type

- [x] Bug

### Specs
What version are you using?
2.17.2

What browser?
Chrome

### Expected Behavior
A custom sortFn function is used for sorting.

### Actual Behavior
The default comparison (column.typeDef.compare function) is used when 0 is returned from a custom sortFn function.
This is caused by a bug in the following lines of code in Table.vue

```
//* if a custom sort function has been provided we use that
              const { sortFn } = column;
              if (sortFn && typeof sortFn === 'function') {
                sortValue =
                  sortValue ||
                  sortFn(xvalue, yvalue, column, xRow, yRow) *
                    (this.sorts[i].type === 'desc' ? -1 : 1);
              }
              //* else we use our own sort
              sortValue =
                sortValue ||
                column.typeDef.compare(xvalue, yvalue, column) *
                  (this.sorts[i].type === 'desc' ? -1 : 1);
```

When sortValue is 0, javascript uses type coercion and converts it to false in the last line.
This means that two values that should be equal are incorrectly sorted by the default comparison.
This throws errors when using a field that isn't the correct type and trying to do a custom sort/format/filter.

This can be fixed by either explicitly checking for null, or changing the code to:

```
//* if a custom sort function has been provided we use that
              const { sortFn } = column;
              if (sortFn && typeof sortFn === 'function') {
                sortValue =
                  sortValue ||
                  sortFn(xvalue, yvalue, column, xRow, yRow) *
                    (this.sorts[i].type === 'desc' ? -1 : 1);
              }
              //* else we use our own sort
              else
              {
               sortValue =
                sortValue ||
                column.typeDef.compare(xvalue, yvalue, column) *
                  (this.sorts[i].type === 'desc' ? -1 : 1);
              }
```

### Steps to Reproduce the Problem

Create a sortFn that only returns 0 for a simple reproduction.
See fiddle below for a more complex example.

### jsfiddle
https://jsfiddle.net/0ue37wm5/1/
Sort by age and the error "d.toLowerCase is not a function" will be thrown as the table tries to use the default comparison.

**Comment by xaksis** [comment]
hey @SethRogers7420 I see your point however in your fiddle, we're explicitly saying that data is text, while supplying numbers? I wonder if this necessarily should throw an error (instead of implicit conversion) to let you know you're doing something wrong here. 
Imagine if we we specified date type and provided arbitrary strings. Table will throw an error saying cannot format arbitrary string as date.

**Comment by SethRogers7420** [comment]
The title change isn't really correct, this bug occurs regardless of what the type of the column is. The fiddle was just to demonstrate this.

Supplying any custom sort function that returns 0 will fall back to the default sort, which shouldn't be happening since the sort function is supposed to return -1, 0, or 1.

My actual use case with this is a column that displays a date range, like 3/15/2019 - 3/20/2018, which requires a custom sort, format, and filter function. Using an actual string as the field is somewhat problematic because the filterFn function doesn't pass in the row, so I have to parse the string back into a date to sort it.

**Comment by xaksis** [comment]
Ohhhh! wow! I totally didn't understand your question when I first read it. Yes, needs an else there... will fix! Thank you so much for clarifying.

**Comment by SethRogers7420** [comment]
Thank you very much.

**Comment by xaksis** [comment]
fixed in v2.18
